src/squeak/ids.js

```diff
@@ -1,50 +1,181 @@
+     * samples in the sequence. */
     SOUND: 12,
+
+    /** A sequence of 32 bit color values. The next 4 bytes defines the number
+     * of colors in the bitmap. */
     BITMAP: 13,
+
+    /** A utf8 string value. The next 4 bytes defines the number of bytes used
+     * by the string. */
     UTF8: 14,
+
+    /** An array header. The next 4 bytes defines the following number of
+     * fields in the array. */
     ARRAY: 20,
+
+    /** An array header. The next 4 bytes defines the following number of
+     * fields in the array. */
     ORDERED_COLLECTION: 21,
+
+    /** An array header. The next 4 bytes defines the following number of
+     * fields in the array. */
     SET: 22,
+
+    /** An array header. The next 4 bytes defines the following number of
+     * fields in the array. */
     IDENTITY_SET: 23,
+
+    /** A dictionary header. The next 4 bytes defines the following number of
+     * key/value field pairs in the dictionary. */
     DICTIONARY: 24,
+
+    /** A dictionary header. The next 4 bytes defines the following number of
+     * key/value field pairs in the dictionary. */
     IDENTITY_DICTIONARY: 25,
+
+    /** A color value. The next 4 bytes represents the color. */
     COLOR: 30,
+
+    /** A color value. The next 4 bytes represents the red, green, and blue
+     * subpixels. The following byte represents the alpha. */
     TRANSLUCENT_COLOR: 31,
+
+    /** A 2 field header. The next 2 fields are the x and y values of this
+     * point. */
     POINT: 32,
+
+    /** A 4 field rectangle header. The next 4 fields are the x, y, x2, y2
+     * values of this rectangle. */
     RECTANGLE: 33,
+
+    /** A 5 field image header. The next 5 fields are the width, height, bit
+     * depth, unused, and bytes. */
     FORM: 34,
+
+    /** A 6 field image header. The next 6 fields are the width, height, bit
+     * depth, unsued, bytes and colormap. */
     SQUEAK: 35,
+
+    /** An object reference to a position in the top level array of fields in a
+     * block. */
     OBJECT_REF: 99,
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object. */
     MORPH: 100,
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object. */
     ALIGNMENT: 104,
-    // Called String in Scratch 2. To reduce confusion this is called
-    // STATIC_STRING to differentiate it from STRING in this codebase.
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object.
+     *
+     * In Scratch 2 this is called String. To reduce confusion in the set of
+     * types, this is called STATIC_STRING in this converter. */
     STATIC_STRING: 105,
+
+    /** A variable field header. The next byte is the version of this object.
```

Comment:
Is this one and are the ones below done or are they TODOs?

Reply:
They are done. The descriptions I'm writing in this file is only meant to be the raw data in the `.sb` file stored in Field instances. As this comment states, this header, appears as two bytes in the `.sb` file, a version byte, and a size byte. I copied the comment as the comment will appear in the produced documentation that way. Everything after `99` works that way.

Hmm. I could move this byte information into the FieldObjectHeader and use `{@link FieldObjectHeader}` to make this shorter but retain that information.